### ISSUE TITLE
this.props is always empty

### ISSUE THREAD

**Issue by iamtmrobinson** [rackt/react-router#1422]
I've followed a [couple][1] of [examples][2] in an attempt to get access to a parameter from a Route in the React component that handles it. However the result of `console.log` on `this.props` from inside the `render` or `componentDidMount` is always `{}` when I'd expect it to contain the `gameId` from the `gamesView` route.

`client.js` which starts the Router:

    // HTML5 History API fix for local 
    if (config.environment === 'dev') {
        var router = Router.create({ routes: routes });
    } else {
        var router = Router.create({ routes: routes, location: Router.HistoryLocation });
    }

    router.run(function(Handler) {
        React.render(
            <Handler />,
            document.getElementById('app')
        );
    });

`routes.js` with some routes removed for simplicity:

    var routes = (
        <Route name='home' path='/' handler={app}>
            <DefaultRoute handler={home} location="/" />
            <Route name='gamesView' path='/games/:gameId' handler={gamesView} />
        </Route>
    );

    module.exports = routes;

...and `app.js` which wraps the other routes, I've tried it both with and without `{...this.props}` in the `RouteHandler`. If I `console.log(this.props)` from inside the `render` function here is also returns `{}`:

    var App = React.createClass({
        render: function() {
            return (
                <div className='container'>
                    <div className='row'>
                        <RouteHandler {...this.props} />
                    </div>
                </div>
            );
        }
    });

    module.exports = App;

Finally the `gamesView` React component that I expect to see the props object. Here `this.props` is also `{}` and the following results in the error `TypeError: $__0 is undefined     var $__0=    this.props.params,gameId=$__0.gameId;`:

    var GamesView = React.createClass({
        render: function() {
            var { gameId } = this.props.params;

            return (
                <div>
                    <h1>Game Name</h1>
                    <p>{gameId}</p>
                </div>
            );
        }
    });

    module.exports = GamesView;

Anybody have any ideas?

  [1]: http://rackt.github.io/react-router/#Router%20Overview
  [2]: https://github.com/rackt/react-router/blob/master/examples/query-params/app.js

**Comment by taurose** [comment]
You linked to the new (1.0 beta) examples. See https://github.com/rackt/react-router/blob/v0.13.3/examples/query-params/app.js

Still, your code should work as far as I can tell. Are you using any wrappers / HoC for your route handlers?

**Comment by Schniz** [comment]
@taurose I don't think it could work on react router<1.0.
tl;dr: use the `context` for RR<1.0.

Your situation can be fixed, however. using a `Root` component or something.
Pay attention that if you use `props` you have to pass it for every `RouteHandler` and its not being "magically injected" using `context`.


Pseudo code (that might work):

```js
var { RouteHandler } = require('react-router');
class Root extends React.Component {
  static contextTypes = {
    router: React.PropTypes.func,
  };
  render() {
    return (<RouteHandler {...this.props} params={ this.context.router.getCurrentParams() } />);
  }
}
```
or something like that.

**Comment by Schniz** [comment]
oh and `params` may be in the router state (`Router.run(...., (Handler, state) => { })`) - you can test it out.
That could fix your problem without an extra component

**Comment by taurose** [comment]
> I don't think it could work on react router<1.0.
> tl;dr: use the context for RR<1.0.

Wasn't sure about that either, but it's definitely in 0.13.3: https://github.com/rackt/react-router/blob/master/CHANGELOG.md

**Comment by iamtmrobinson** [comment]
> You linked to the new (1.0 beta) examples.
> @taurose 
This tipped me off, I hadn't realised that at all. I'm on a relatively old version of 0.11.6 where the [docs show](https://github.com/rackt/react-router/blob/v0.11.6/examples/query-params/app.js) use of the `Router.State` mixin and then getting the expected param via `var gameId = this.getParams().gameId;` which worked for me.

Thanks for the help guys, I'll also have a look into updating to a newer version of RR.
